homeassistant/components/automation/__init__.py

```diff
@@ -460,7 +460,11 @@ async def async_trigger(self, run_variables, context=None, skip_condition=False)
         ) as automation_trace:
             if self._variables:
```

Comment:
We should always set the `this` variable:
```diff
+            this = self.hass.states.get(self.entity_id).as_dict()
+            run_variables = {"this": this, **run_variables}
             if self._variables:
                 try:
-                    states = self.hass.states.get(self.entity_id)
-                    this = states.as_dict() if states else {"entity_id": self.entity_id}
                     variables = self._variables.async_render(
-                        self.hass, {"this": this, **run_variables}
+                        self.hass, run_variables
                     )
```

Reply:
Changed